### ISSUE TITLE
Improve documentation for System.Threading.Channels

### ISSUE THREAD

**Issue by momt99** [dotnet/runtime#59617]
I think that API documentation for System.Threading.Channels classes are quite abstract and short and do not include the necessary information.

For example, consider `TryWrite` method of `ChannelWriter`. It is only noted that if it successfully writes to the channel, the result will be true. OK, but when it is unsuccessful? In my case, I had to take a look at blog posts and the source code to realize that for an unbounded channel the result is always true unless the channel is completed while for a bounded one it may return false for other additional scenarios.

I'm not sure if I'm right. Maybe as a purpose of abstraction, we should not be aware of these behaviors at all, but I believe that something is missing here and these details should be included somewhere either in remarks or any other appropriate place.

**Comment by carlossanlop** [comment]
Thanks for reporting. I'm glad you could clarify some of your questions by reading the blog post, they are part of our documentation as well. I assume you're talking about the official blog post and video?:

- Blog post: https://devblogs.microsoft.com/dotnet/an-introduction-to-system-threading-channels/
- Video: https://channel9.msdn.com/Shows/On-NET/Working-with-Channels-in-NET

We can definitely improve the MS Docs documentation with the info you found in the blog post. And external contributions are also welcome: If you go to the MS Docs page for the API you want to improve, click on Edit on the top right, then you can submit a PR to dotnet-api-docs to modify the xml containing the docs. The information you see there is what we publish in our IntelliSense.

**Comment by momt99** [comment]
Thanks for the explanation. Indeed, the authors are more aware of the details and the essential points, and the best ones to complete the docs. However, I won't hesitate to help if I find an opportunity to help.

**Comment by stephentoub** [comment]
> If the Channel is full, and we're in DropWrite mode, surely it stands to reason that a false Boolean would help inform users of the API that the item was not written to the Channel (despite the operation to try having succeeded, and resulted in a dropped message)?

No, the write was successful, with the channel then dropping an element, and choosing based on the selected policy to drop the one that was just written.

**Comment by enclave-marc-barry** [comment]
Not to nit-pick, but in `DropWrite` mode, I'm not sure it we actually are completing any kind of a write operation? https://github.com/dotnet/runtime/blob/main/src/libraries/System.Threading.Channels/src/System/Threading/Channels/BoundedChannel.cs#L423-L428

Regardless, it would still be helpful to update the documentation if possible.

**Comment by stephentoub** [comment]
> I'm not sure we actually are completing any kind of write and drop operation?

We are. The item is passed to the channel, which accepts it, declares that it was written, and then has to drop one, and based on policy, it selects the one just written. The actual way that's achieved is an implementation detail.

> Regardless, it would still be helpful to update the documentation if possible.

Yup, thanks.
